sdk/src/azure/iot/az_iot_hub_client_twin.c

```diff
@@ -173,8 +173,9 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_parse_received_topic(
       {
         // Is a reported prop response
         out_response->response_type = AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES;
-        _az_RETURN_IF_FAILED(az_iot_message_properties_find(
-            &props, az_iot_hub_twin_version_prop, &out_response->version));
+        out_response->version = AZ_SPAN_EMPTY;
+        az_iot_message_properties_find(
+            &props, az_iot_hub_twin_version_prop, &out_response->version);
```

Comment:
We should still check if this fails, and if it does with a `AZ_ERROR_ITEM_NOT_FOUND`, explicitly set `&out_response->version = AZ_SPAN_EMPTY`. Wouldn't want garbage going in there by accident.

Reply:
Also...you're going to hate me.....a test with this scenario :)

Reply:
Thanks for commenting!  I was curious to know what others were thinking on this issue.  We will need to do a test with IoT Edge, not IoT Hub.  Do you know how we go about creating devices that talk to IoT Edge and not Hub?

Reply:
That makes sense to switch the order to look only for the specific AZ failure of version being absent.

Reply:
> Thanks for commenting! I was curious to know what others were thinking on this issue. We will need to do a test with IoT Edge, not IoT Hub. Do you know how we go about creating devices that talk to IoT Edge and not Hub?

A unit test should be enough for it.

Reply:
Side note: The underlying az_iot_message_properties_find only returns AZ_OK or AZ_ERROR_ITEM_NOT_FOUND, so for now there isn't another failure option.  However, if the logic ever changed in the future to include a different possible error return, we would be missing that scenario if we don't add the check in at this level.